index.js

```diff
@@ -78,22 +80,25 @@ class CssWriter {
 	}
 }
 
-/** @returns {import('rollup').Plugin} */
+/**
+ * @param [options] {Partial<import('.').Options>}
+ * @returns {import('rollup').Plugin}
+ */
 module.exports = function (options = {}) {
-	const extensions = options.extensions || ['.svelte'];
-	const filter = createFilter(options.include, options.exclude);
+	const { compilerOptions={}, ...rest } = options;
+	const extensions = rest.extensions || ['.svelte'];
+	const filter = createFilter(rest.include, rest.exclude);
 
-	/** @type {import('svelte/types/compiler/interfaces').ModuleFormat} */
-	const format = 'esm', config = { format };
+	compilerOptions.format = 'esm';
+	const isDev = !!compilerOptions.dev;
 
-	for (let key in options) {
-		// forward `svelte/compiler` options
+	for (let key in rest) {
 		if (plugin_options.has(key)) continue;
-		config[key] = config[key] || options[key];
+		console.warn(`${PREFIX} Unknown "${key}" option. Please use \`compilerOptions\` for any Svelte compiler configuration.`);
```

Comment:
I'd expect single quotes or double quotes around `compilerOptions` rather than escaped backticks since backticks don't have any effect on the command line like they do in markdown

Reply:
Updated 👍 Left the ones around `package.json`
TBH just used backticks because that's what was already there.